gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/JWKSFederationFilter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.knox.gateway.provider.federation.jwt.filter;
+
+import java.io.IOException;
+import java.text.ParseException;
+
+import javax.security.auth.Subject;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.knox.gateway.services.security.token.impl.JWT;
+import org.apache.knox.gateway.services.security.token.impl.JWTToken;
+
+public class JWKSFederationFilter extends AbstractJWTFilter {
```

Comment:
Maybe I am missing something but why do we need a new filter? The existing filter can just have the new parameters potentially?

Reply:
@risdenk  I don't want to modify existing filter as it may cause confusion for JWTProvider parameters , if someone put both JWKS url  and Publiccert in topology , it is tough to explain . this provider clearly dedicated to JWKS url.. no public cert . its my point of view but if community suggest to modify  same filter i will update same. Please let me know

Reply:
So right now that logic of what is chosen is in AbstractJwtFilter. Its pretty clear that if a public key is provided it will be used over URL. Having both is definitely in valid. Not sure why we would need a separate filter. The URL is a separate alternative option to the public key.

Reply:
removed  new filter , modified existing JWTfilter . Need to add another http header  for Hive JDBC where we can't pass token with Bearer  keyword.